core/src/main/java/com/scalar/db/api/DistributedTransactionManager.java

```diff
@@ -196,6 +196,18 @@ DistributedTransaction start(String txId, SerializableStrategy strategy)
   DistributedTransaction start(String txId, Isolation isolation, SerializableStrategy strategy)
       throws TransactionException;
 
+  /**
+   * Joins an ongoing transaction associated with the specified transaction ID.
+   *
+   * @param txId the transaction ID
+   * @return {@link DistributedTransaction}
+   * @throws TransactionNotFoundException if the transaction associated with the specified
+   *     transaction ID is not found. You can retry the transaction from the beginning
+   */
```

Comment:
The javadoc for `join()` and `resume()` is almost identical so I am worried uses might overlook in which case they use `join()`. I assume you would like users to use join in the scenario you described [here](https://github.com/scalar-labs/scalardb/pull/944#issuecomment-1641462063) so should the `join()` documentation be more specific and have a similar wording as the [join() javadoc for 2pc](https://github.com/scalar-labs/scalardb/blob/e5bf6ebda1baba295273447d17cbb0c49fdd974b/core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java#L125C1-L126C101)

```suggestion
  /**
   * Joins an ongoing transaction associated with the specified transaction ID. The transaction should be
   * started by a coordinator process. This method is assumed to be called by a participant process.
   *
   * @param txId the transaction ID
   * @return {@link DistributedTransaction}
   * @throws TransactionNotFoundException if the transaction associated with the specified
   *     transaction ID is not found. You can retry the transaction from the beginning
   */
```
Or maybe, you are keeping the javadoc similar to `resume()` for a reason?

Reply:
That's a possible idea, but in the context of `DistributedTransaction`, the terms `coordinator` and `participant` are not defined, so users may get confused.

Actually, the primary use case of this method `join()` is intended for ScalarDB Cluster. Therefore, I think we should provide some detailed documentation for it on the ScalarDB Cluster side. As for the ScalarDB core side, it might be best to keep a simpler explanation. What do you think?

Reply:
That makes sense; let's keep it like this. hank you.